adaptive_hci/controllers.py

```diff
@@ -1,18 +1,81 @@
+import re
 import lightning.pytorch as pl
 import torch
-from torch.functional import F
+import torch.nn as nn
+from torch.nn import functional as F
 from torchmetrics import ExactMatch, F1Score, Accuracy
 from vit_pytorch import ViT
 
 
 class EMGViT(ViT):
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
+    def __init__(self, config, *args, **kwargs):
+        super().__init__(image_size=config.window_size,
+            patch_size=config.base_model.patch_size,
+            num_classes=config.num_classes,
+            dim=config.base_model.dim,
+            depth=config.base_model.depth,
+            heads=config.base_model.heads,
+            mlp_dim=config.base_model.mlp_dim,
+            dropout=config.base_model.dropout,
+            emb_dropout=config.base_model.emb_dropout,
+            channels=config.base_model.channels,
+            *args, **kwargs)
 
     def __call__(self, x: torch.Tensor):
         x.unsqueeze_(axis=1)
         return self.forward(x)
 
+    def freeze_layers(self, n_frozen_layers: int) -> None:
```

Comment:
I feel this code is duplicated, can you make a function that works for arbitrary networks?